documentation/ssr-and-hydration.mld

```diff
@@ -0,0 +1,291 @@
+{0 SSR and hydration}
+
+{1 How does React handle SSR markup?}
+During server-side rendering (SSR), React generates static HTML markup on the server. When this markup reaches the client's browser, React performs a process called "hydration", 
+it attaches event handlers and other interactive behaviors to the existing HTML elements, rather than re-rendering the entire page from scratc
+
+For hydration to work correctly, the initial server markup must match exactly what React expects on the client - otherwise React will be forced to re-render the page, negatively impacting performance.
+
+For example:
+- server renders:[<span>Hello from server</span>]
+- client renders:[<span>Hello from client</span>]
+This mismatch will trigger a hydration error. Instead of being able to efficiently "hydrate" the existing server markup with interactivity, React must throw away the server-rendered content and do a full client-side render, resulting in worse performance and potential visual flickering.
+
+Read more about it in the [React documentation](https://react.dev/reference/react-dom/client/hydrateRoot).
+
+{1 Hydrate error}
+
+Commonly, hydrate errors will appear as:
+{%html:
+  <p style="background-color: #362422; color: #df5452; padding: 1em;">
+    <strong>Text content does not match server-rendered HTML.</strong>
+  </p>
+%}
+
+This means that while rendering the application, there was a difference between the React tree pre-rendered from the server and the React tree rendered during the first render in the browser (hydration).
+
+As it's hard to see it just in words, let's go to an example with this simple code:
+
+{@reasonml[
+[@react.component]
+let make = () => {
+  let backgroundColor = switch%platform () {
+    | Server => "red"
+    | Client => "blue"
+  };
+
+  <div className=backgroundColor />;
+};
+]}
+
+To understand how [switch%platform] works, look at the {{!page-"browser_ppx"} browser_ppx page}
+And let's see the diagram above, which shows how the code will behave on each SSR step:
+
+{%html:<img src="./ssr-hydrate-issue.png" alt='Hydrate error' style="max-width: 100%; height: auto; vertical-align: middle;">%}
+
+* As we can see, there is a difference between III and IV, that will cause React to complain about content mismatch.
+
+This is because React understands that the markup from the server ([background-color: red](IV)) differs from the one it expects to be on the client at the hydration time ([background-color: blue](V)). 
+
+If we want different values on the server and client, we must do it after React hydrates the client or, commonly said, after *mount*.
+
+To work on it we must check if the client mounted and then change this value. This can be done with a hook that will be executed only after the client is hydrated.
```

Comment:
Add a heading like, "Possible solution: mounting on the client only"

Reply:
and mention that ideally those hydration errors should be fixed, either by removing the difference or by adding this extra solutions